project1/src/main/java/uk/co/cpsd/javaproject1/WorldPanel.java

```diff
@@ -39,6 +40,23 @@ protected void paintComponent(Graphics g) {
             
             g.setColor(animal.getColor());
             g.fillRect(animal.getX() * cellSize+10, animal.getY() * cellSize+10, cellSize-15,cellSize-15);
+
+            
+            // Goats have number id, and is shown in the panel. Code has been written by AI
+
+            g.setColor(Color.BLACK); // Set color for the text (e.g., black)
+            // Adjust font size based on cell size for better visibility
+            g.setFont(new Font("Arial", Font.BOLD, Math.max(10, cellSize / 3))); // <--- NEW: Set font
```

Comment:
You're creating a new Font object every time you update a square.  But they are all identical fonts.  You should create one (as a static final object) and share it